OrbitGl/LiveFunctionsDataView.cpp

```diff
@@ -193,10 +196,14 @@ void LiveFunctionsDataView::OnContextMenu(const std::string& action, int menu_in
       GOrbitApp->DeselectFunction(*function);
     }
   } else if (action == kMenuActionDisassembly) {
-    int32_t pid = GOrbitApp->GetCaptureData().process_id();
+    const CaptureData& capture_data = GOrbitApp->GetCaptureData();
+    int32_t pid = capture_data.process_id();
     for (int i : item_indices) {
       const FunctionInfo& function = *GetFunction(i);
-      GOrbitApp->Disassemble(pid, function);
+      const uint64_t absolute_address = FunctionUtils::GetAbsoluteAddress(function);
+      if (capture_data.FindModuleByAddress(absolute_address) != nullptr) {
+        GOrbitApp->Disassemble(pid, function);
```

Comment:
What I don't understand here is the relationship between the condition and the action&mdash;while conceptually clear, it is actually very loose. If I understand correctly `Disassemble` doesn't rely on anything related to modules as stored in the client, rather it just tries to contact the remote process and queries its memories. This can be seen in the fact that `FindModuleByAddress` is only based on `absolute_address`, while `Disassemble` uses pid&mdash;what happens if for a coincidence `absolute_address` makes sense but `pid` is not available? Would that cause a crash?
So this actually brings me to another question: Can you explain where the crash was? How does this condition relate to the crash?

Reply:
So okay, let me start by what was the crash here (in fact there were also other crashes).
`pid` was used in order to check if the process is 32 or 64 bit. That process, did not exists, so there was a null deference. 
`CaptureData::FindModuleByAddress` uses also the `process` to find the module. This is the exact same process, where we get the `pid` from. So they can't be off sync.

Also note, that since the recent refactoring the `ModuleData*` will be also present if the module is not loaded. So it might be only null, if we are in a loaded capture (or if we would implement unloading of modules in the future).

However, I see that using the module here to check that we can access the memory of a function is a bit confusing.
In the other views (and also already in the topdownview), the `CaptureData::FindFunctionByAddress` is used, which returns a `FunctionInfo *`. However, as we already have a `FunctionInfo *` here (as we actually reconstruct function infos for the instrumented functions that are shown here), I would find it more confusing to use an additional `FunctionInfo* `

Reply:
Orbit crashed when the `GOrbitApp->Disassemble` is called without being connected to a remote. Which somewhat makes sense, since the memory cannot be retrieved from remote. 

capture_data does apparently not contain module information when the capture was loaded from a file, hence this actually works. 

I would prefer something else than this "if module available". Preferably some kind of CHECK somewhere when disassembly is attempted without being connected (i dont think this code location is actually suited). A CHECK should be good idea, because we already make sure that disassembly is possible when activating the context menu. 

If you want to go with how it currently is, please at least LOG or ERROR something, and maybe a comment whats happening is a good idea

Reply:
Well the actual crash was already before the attempt to load the memory, when trying to get the bittness of the process.
I added a CHECK there. In case the process exists but the remote process fails, there is already an error message appearing to the user.

Reply:
So, for the crash in `OrbitApp::Disassemble` caused by `data_manager_->GetProcessByPid(pid)` being `nullptr`, I think that this case should be handled gracefully directly there. I think that other crash situations like not being connected to a remote should also first of all be handled gracefully.
But obviously in general we should not provide the option when the action is not possible (and not just give an error afterward). Maybe we could have some kind of `OrbitApp::CanDisassemble` that checks the connection first?
What do you think?

Reply:
So instead of that CHECK in Disassemble, you would have an `if(process==nullptr) SendErrorToUI`? And then have a `CanDisassemble` Method in App?
In general I like the idea. But I am quite unsure of how this `CanDisassemble` method should look like.